src/test/java/org/dataone/hashstore/HashStoreTest.java

```diff
@@ -1,18 +1,351 @@
+
+        // Create a thread pool with 2 threads
+        ExecutorService executorService = Executors.newFixedThreadPool(2);
+
+        // Submit 2 threads, each calling storeObject
+        Future<?> future = executorService.submit(() -> {
+            try {
+                InputStream dataStream = new FileInputStream(testDataFile);
+                HashAddress objInfo = hsj.storeObject(dataStream, pid, null, null, null);
+                if (objInfo != null) {
+                    String absPath = objInfo.getAbsPath();
+                    File permAddress = new File(absPath);
+                    assertTrue(permAddress.exists());
+                }
+            } catch (Exception e) {
+                fail("future - Unexpected Exception: " + e.getMessage());
+            }
+        });
+        Future<?> future_dup = executorService.submit(() -> {
+            try {
+                InputStream dataStreamDup = new FileInputStream(testDataFile);
+                HashAddress objInfoDup = hsj.storeObject(dataStreamDup, pid, null, null, null);
+                if (objInfoDup != null) {
+                    String absPath = objInfoDup.getAbsPath();
+                    File permAddress = new File(absPath);
+                    assertTrue(permAddress.exists());
+                }
+            } catch (Exception e) {
+                fail("future_dup - Unexpected Exception: " + e.getMessage());
+            }
+        });
+
+        // Wait for all tasks to complete
+        executorService.shutdown();
+        executorService.awaitTermination(1, TimeUnit.MINUTES);
+    }
+
+    /**
+     * Check store object pid lock for duplicate object exception
+     * 
+     * Two futures (threads) will run concurrently, one of which will encounter an
+     * ExecutionException. The future that yields the exception is then checked to
+     * confirm that a FileAlreadyExistsException exception is thrown.
+     */
+    @Test
+    public void testStoreObjectObjectLockedIdsFileExistsException() throws Exception {
+        // Get test file to "upload"
+        String pid = "jtao.1700.1";
+        Path testdataDirectory = Paths.get("src/test/java/org/dataone/hashstore", "testdata", pid);
+        String testdataAbsolutePath = testdataDirectory.toFile().getAbsolutePath();
+        File testDataFile = new File(testdataAbsolutePath);
+
+        // Create a thread pool with 2 threads
+        ExecutorService executorService = Executors.newFixedThreadPool(2);
+
+        // Submit 2 threads, each calling storeObject
+        Future<?> future = executorService.submit(() -> {
+            try {
+                InputStream dataStream = new FileInputStream(testDataFile);
+                HashAddress objInfo = hsj.storeObject(dataStream, pid, null, null, null);
+            } catch (Exception e) {
+                Class<?> exceptionClass = e.getClass();
+                assertTrue(exceptionClass == FileAlreadyExistsException.class);
```

Comment:
can replace 2 lines with
`assertTrue(e instanceof FileAlreadyExistsException);`

Reply:
Thank you - this is way cleaner!